application/comit_node/src/http_api/action.rs

```diff
@@ -0,0 +1,302 @@
+                        data_type: "float",
+                        description:
+                        "The fee-per-byte you want to pay for the redeem transaction in satoshis.",
+                    },
+                ],
+            )),
+        }
+    }
+}
+
+impl ListRequiredFields for bitcoin::SpendOutput {
+    fn list_required_fields() -> Vec<siren::Field> {
+        vec![
+            siren::Field {
+                name: "address".to_owned(),
+                class: vec!["bitcoin".to_owned(), "address".to_owned()],
+                _type: Some("text".to_owned()),
+                value: None,
+                title: None,
+            },
+            siren::Field {
+                name: "fee_per_byte".to_owned(),
+                class: vec!["bitcoin".to_owned(), "feePerByte".to_owned()],
+                _type: Some("number".to_owned()),
+                value: None,
+                title: None,
+            },
+        ]
+    }
+}
+
+impl IntoResponsePayload for ethereum::ContractDeploy {
+    fn into_response_payload(
+        self,
+        query_params: ActionExecutionParameters,
+    ) -> Result<ActionResponseBody, HttpApiProblem> {
+        let ethereum::ContractDeploy {
+            data,
+            amount,
+            gas_limit,
+            network,
+        } = self;
+        match query_params {
+            ActionExecutionParameters::None {} => Ok(ActionResponseBody::EthereumDeployContract {
+                data,
+                amount,
+                gas_limit,
+                network,
+            }),
+            _ => Err(problem::unexpected_query_parameters(
+                "ethereum::ContractDeploy",
+                vec!["address".into(), "fee_per_byte".into()],
+            )),
+        }
+    }
+}
+
+impl ListRequiredFields for ethereum::ContractDeploy {
+    fn list_required_fields() -> Vec<siren::Field> {
+        vec![]
+    }
+}
+
+impl IntoResponsePayload for ethereum::CallContract {
+    fn into_response_payload(
+        self,
+        query_params: ActionExecutionParameters,
+    ) -> Result<ActionResponseBody, HttpApiProblem> {
+        let ethereum::CallContract {
+            to,
+            data,
+            gas_limit,
+            network,
+            min_block_timestamp,
+        } = self;
+        match query_params {
+            ActionExecutionParameters::None {} => Ok(ActionResponseBody::EthereumCallContract {
+                contract_address: to,
+                data,
+                gas_limit,
+                network,
+                min_block_timestamp,
+            }),
+            _ => Err(problem::unexpected_query_parameters(
+                "ethereum::SendTransaction",
+                vec!["address".into(), "fee_per_byte".into()],
+            )),
+        }
+    }
+}
+
+impl ListRequiredFields for ethereum::CallContract {
+    fn list_required_fields() -> Vec<siren::Field> {
+        vec![]
+    }
+}
+
+impl ListRequiredFields for Infallible {
```

Comment:
I basically need it because of this:

https://github.com/comit-network/comit-rs/blob/37503aa50c3357219e8dde6b1b4a731c822cff20/application/comit_node/src/swap_protocols/rfc003/alice/actions/generic_impl.rs#L24-L31

`Infallible` is actually specific to conversions at the moment but in general, it represents a type that cannot be instantiated. This is helpful in situation where the types you have modeled in your program force you to put a type but in the actual implementation, it is never instantiated. This can be checked by the compiler at compile-time. `Infalliable` is an empty enum, so there is nothing you can instantiate.

In the long run, Rust will get a `never` type to more generally express this. The `never` type automatically implements all traits and hence, this particular implementation will no longer be necessary. It is a pretty powerful construct :)